syntax = "proto3";

message Shape {
  repeated int32 dim = 1;
}

enum DType {
  UINT8 = 0;
  INT32 = 1;
  FLOAT32 = 2;
}

message ItemProto {
  int32 length = 1;
  repeated Shape shape = 2;
  repeated bytes data = 3;
  repeated DType dtype = 4;
}

message StepProto {
  ItemProto observation = 1;
  ItemProto action = 2;
  ItemProto reward = 3;
  float terminal = 4;
  bool timeout = 5;
  int32 rollout_id = 6;
}

message StepReply {
  string status = 1;
}

service StepService {
  rpc Send(StepProto) returns (StepReply);
}
